src/Microsoft.Extensions.DependencyInjection/ServiceLookup/CallSiteValidator.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
+{
+    internal class CallSiteValidator: CallSiteVisitor<CallSiteValidator.CallSiteValidatorState, Type>
+    {
+        private readonly Dictionary<Type, Type> _scopedServices = new Dictionary<Type, Type>();
```

Comment:
Comment to indicate the key is the service being resolved via `GetService` and the value is a single scoped service that it depends on (but there may be more.